crates/services/p2p/src/behavior.rs

```diff
@@ -99,20 +104,24 @@ impl<Codec: NetworkCodec> FuelBehaviour<Codec> {
         let peer_report = PeerReportBehaviour::new(p2p_config);
 
         let req_res_protocol =
-            std::iter::once((codec.get_req_res_protocol(), ProtocolSupport::Full));
+            core::iter::once((codec.get_req_res_protocol(), ProtocolSupport::Full));
 
         let mut req_res_config = RequestResponseConfig::default();
-        req_res_config.set_request_timeout(p2p_config.set_request_timeout);
-        req_res_config.set_connection_keep_alive(p2p_config.set_connection_keep_alive);
+        req_res_config
+            .clone()
+            .with_request_timeout(p2p_config.set_request_timeout);
+
+        // TODO: Do we need to set this via the `SwarmBuilder` somewhere? https://github.com/libp2p/rust-libp2p/pull/4679
+        // req_res_config.set_connection_keep_alive(p2p_config.set_connection_keep_alive);
```

Comment:
`with_idle_connection_timeout` is only on the Swarm `Config`, so it would probably have the same effect as using the method on the `SwarmBuilder`. There are the other comments here about using the `SwarmBuilder` so I assume I'd go with whichever method, the builder method or the config method depending on what we land on there.

Reply:
Yes, `SwarmBuilder` has a `with_swarm_config` function that allows you to modify the config. If you construct `Swarm` directly then you can just construct the config manually yes.